pkg/cmd/root/root.go

```diff
@@ -72,7 +72,11 @@ func NewCmdRoot(f *cmdutil.Factory, version, buildDate string) (*cobra.Command,
 		PersistentPreRunE: func(cmd *cobra.Command, args []string) error {
 			// require that the user is authenticated before running most commands
 			if cmdutil.IsAuthCheckEnabled(cmd) && !cmdutil.CheckAuth(cfg) {
-				fmt.Fprint(io.ErrOut, authHelp())
+				if cmd.Parent().Use == "codespace" {
+					fmt.Fprint(io.ErrOut, "To get started with GitHub CLI, please run:  gh auth login -s codespace")
+				} else {
+					fmt.Fprint(io.ErrOut, authHelp())
+				}
```

Comment:
Minor suggestion regarding output formatting:

```suggestion
				if cmd.Parent().Use == "codespace" {
					fmt.Fprintln(io.ErrOut, "To get started with GitHub CLI, please run:  gh auth login -s codespace")
				} else {
					fmt.Fprint(io.ErrOut, authHelp())
				}
```

Beyond that, I think it's a little weird the exit code differs between the 2 test cases below of `1` _(error)_ versus `4` _(auth)_

## Manual testing

### Test 1: Completely unauthenticated

```shell
andyfeller@Andrews-MacBook-Pro:cli/cli ‹hintAddingCodeSpaceScopeToAuthLogin*›$ gh auth logout
✓ Logged out of github.com account 'andyfeller'

andyfeller@Andrews-MacBook-Pro:cli/cli ‹hintAddingCodeSpaceScopeToAuthLogin*›$ go run cmd/gh/main.go codespace list 
To get started with GitHub CLI, please run:  gh auth login -s codespaceexit status 4
```

### Test 2: Authenticated without `codespaces` scope

```shell
andyfeller@Andrews-MacBook-Pro:cli/cli ‹hintAddingCodeSpaceScopeToAuthLogin*›$ gh auth login --web -h github.com
? What is your preferred protocol for Git operations? HTTPS

! First copy your one-time code: 9581-6020
Press Enter to open github.com in your browser... 
✓ Authentication complete.
- gh config set -h github.com git_protocol https
✓ Configured git protocol
✓ Logged in as andyfeller

andyfeller@Andrews-MacBook-Pro:cli/cli ‹hintAddingCodeSpaceScopeToAuthLogin*›$ go run cmd/gh/main.go codespace list 
error getting codespaces: HTTP 403: Must have admin rights to Repository. (https://api.github.com/user/codespaces?per_page=30)
This API operation needs the "codespace" scope. To request it, run:  gh auth refresh -h github.com -s codespace
exit status 1

```

Reply:
@andyfeller though the exit status for both the test cases are different but those correctly represent the error state. Let me know if you have any suggestion here.

Reply:
@rajhawaldar : all good, just something I was noticing 👍